inc/Engine/Optimization/RUCSS/Warmup/Status/RESTWP.php

```diff
@@ -0,0 +1,167 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Optimization\RUCSS\Warmup\Status;
+
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Engine\Optimization\RUCSS\Database\Queries\ResourcesQuery;
+
+/**
+ * Class RESTWP
+ *
+ * @package WP_Rocket\Engine\Optimization\RUCSS\Warmup\Status
+ */
+class RESTWP {
+
+	/**
+	 * Namespace for REST Route.
+	 */
+	const API_NAMESPACE = 'wp-rocket/v1';
+
+	/**
+	 * Part of route namespace for this inherited class item type.
+	 *
+	 * @var string $route_namespace to be set with like post, term.
+	 */
+	const API_ROUTE = 'rucss/warmup/status';
+
+	/**
+	 * WP Rocket options instance.
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Resources Query instance.
+	 *
+	 * @var ResourcesQuery
+	 */
+	private $resources_query;
+
+	/**
+	 * RESTWP constructor.
+	 *
+	 * @since 3.9
+	 *
+	 * @param Options_Data   $options Instance of options data handler.
+	 * @param ResourcesQuery $resources_query Resources Query instance.
+	 */
+	public function __construct( Options_Data $options, ResourcesQuery $resources_query ) {
+		$this->options         = $options;
+		$this->resources_query = $resources_query;
+	}
+
+	/**
+	 * Registers the generate route in the WP REST API
+	 *
+	 * @since 3.9
+	 *
+	 * @return void
+	 */
+	public function register_status_route() {
+		register_rest_route(
+			self::API_NAMESPACE,
+			self::API_ROUTE,
+			[
+				'methods'             => 'POST',
+				'callback'            => [ $this, 'respond_status' ],
+				'permission_callback' => [ $this, 'check_permissions' ],
```

Comment:
I would convert this to an anonymous method directly

Reply:
u mean check_permissions one?

Reply:
yes, like the following:
 ```
'permission_callback' => function() {
    return current_user_can( 'rocket_remove_unused_css' );
},
```

Reply:
yes, I know the anonymous function for sure but really I hate them :D for many reasons and one of them is to add docblock but it's ok with this one-line method.